OCCommunicationLib/OCCommunicationLib/OCCommunication.m

```diff
@@ -160,44 +161,37 @@ - (void)setSecurityPolicyManagers:(AFSecurityPolicy *)securityPolicy {
 
 #pragma mark - Setting Credentials
 
-
 - (void) setCredentials:(OCCredentialsDto *) credentials {
-
+    
+    self.credDto = credentials;
+    
     switch (credentials.authenticationMethod) {
         case AuthenticationMethodSAML_WEB_SSO:
-            
-            [self setCredentialsWithCookie:credentials.accessToken];
+            self.kindOfCredential = credentialCookie;
```

Comment:
The property ```kindOfCredential```  could be deleted, and ```self.credDto.authenticationMethod``` used instead in any place where the first one is being used now.

Reply:
👍